File: go/remove-element/remove-element.go
package main

import(
	"fmt"
)

func removeElement(nums []int, val int) int{
	result := 0
	for i, n := range nums{
		if n != val{
			nums[result] = nums[i]
			result++
		}
	} 
	return result
} 

func main(){
	nums := []int{3,2,2,3}
	val := 3
	fmt.Println("answer to the first example is:",removeElement(nums, val))
}

File: js/remove-element/remove-element.js
function removeElement(nums, val){
    let result = 0
    for (let i = 0 ; i < nums.length ; i++){
        if (nums[i] != val){
            nums[result] = nums [i]
            result++
        }
    }
    return result
}

let nums = [3,2,2,3]
let val = 3
console.log("answer to the first example is:",removeElement(nums, val))